nvtabular/ops/target_encoding.py

```diff
@@ -25,11 +24,12 @@
     read_parquet_dispatch,
 )
 from merlin.dag import Node
+from merlin.dag.stat_operator import StatOperator
```

Comment:
After this change,  it looks like some ops won't be subclasses of the `nvtabular.ops.Operator` any more. Is the additional `create_node` method and `infererence_initialize` on that base class no longer relevant to the StatOperator? and subsequent operators here? If we'd prefer all operators in NVTabular to inherit from `nvtabular.ops.Operator`, one option might be to change `StatOperator` to a mixin style class that can be used alongside an `Operator` to make it Stat aware (support `fit`). 

```python
class MyOperator(Operator, StatMixin):
    def transform(...):
        ...
    def fit(...):
        ...
```

Reply:
Good catch!

`inference_initialize` is for the custom C++ implementations of the operators for serving; `Categorify` is both a `StatOperator` and has a custom implementation, so that probably does matter.

The `create_node` method specifies a type for Workflows that shims in additional functionality to handle implicit conversion to `LambdaOp`. I'm thinking we should probably move that into Core anyway, since there's nothing particularly NVT-specific about using lambdas to do arbitrary transforms.